cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/sourcecontrol/SourceControlOperationRunnerFactory.java

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright © 2023 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.internal.app.sourcecontrol;
+
+import io.cdap.cdap.sourcecontrol.RepositoryManager;
+
+/**
+ * SourceControlOperatorFactory interface which is used by Guice during runtime.
+ */
+public interface SourceControlOperationRunnerFactory {
```

Comment:
It's not clear to me why we need so many layers. For example, why do we need a runner factory instead of just a runner?

Reply:
@albertshau Removed the `SourceControlOperationRunnerFactory` and added the `RepoManagerFactory` instead. We now only have one runner and inside it creates the `RepoManager` when needed